src/passes/MergeBlocks.cpp

```diff
@@ -434,6 +434,15 @@ struct MergeBlocks
     optimizeBlock(curr, getModule(), getPassOptions(), branchInfo);
   }
 
+  void visitDrop(Drop* curr) {
```

Comment:
What's the difference between the cases handled by this and this code in `optimizeBlock`?
https://github.com/WebAssembly/binaryen/blob/7f30b6c4c0af98642a8f22cbd4e61ff688f9e9fe/src/passes/MergeBlocks.cpp#L262

Can't all cases be handled by this?

Reply:
The code in `optimizeBlock` runs on the children of a block, and we do need to run that more than once, unlike `visitDrop`: Each time we make a signficant change to a block, it's possible that merging with children of the block is possible, so we call `optimizeBlock` on such occasions.

On the other hand, for things whose parent isn't a block, the new `visitDrop` is enough, since the parent can't be merged with it, so we can just do that once.

Reply:
Actually, I was completely wrong here! You were right, we can remove this code. It looks like changes in the parent never lead to new improvements for the child, so we can just do the `visitDrop` part once.

I'll do some fuzzing to verify this.

Reply:
Ah, actually that was wrong, it turns out. There is at least one case where we do end up improving a child based on the parents, which I added a testcase for now (after reverting the last commit). What seems to happen is that after we optimize the parent there, we then look at the child and see that it can be removed with its name. (This was not noticed in the test suite since we run `--remove-unused-names`.)

Logically I still think it makes sense that we only need the call from `visitDrop`, but that would require changes to the pass, and I'm not sure it's worth a large refactoring.